download.sh

```diff
@@ -1,61 +1,130 @@
 #!/bin/bash
 
-# Copyright (c) Meta Platforms, Inc. and affiliates.
-# This software may be used and distributed according to the terms of the Llama 2 Community License Agreement.
-
-read -p "Enter the URL from email: " PRESIGNED_URL
-echo ""
-ALL_MODELS="7b,13b,34b,7b-Python,13b-Python,34b-Python,7b-Instruct,13b-Instruct,34b-Instruct"
-read -p "Enter the list of models to download without spaces ($ALL_MODELS), or press Enter for all: " MODEL_SIZE
-TARGET_FOLDER="."             # where all files should end up
-mkdir -p ${TARGET_FOLDER}
-
-if [[ $MODEL_SIZE == "" ]]; then
-    MODEL_SIZE=$ALL_MODELS
-fi
-
-echo "Downloading LICENSE and Acceptable Usage Policy"
-wget ${PRESIGNED_URL/'*'/"LICENSE"} -O ${TARGET_FOLDER}"/LICENSE"
-wget ${PRESIGNED_URL/'*'/"USE_POLICY.md"} -O ${TARGET_FOLDER}"/USE_POLICY.md"
-
-for m in ${MODEL_SIZE//,/ }
-do
-    case $m in
-      7b)
-        SHARD=0 ;;
-      13b)
-        SHARD=1 ;;
-      34b)
-        SHARD=3 ;;
-      7b-Python)
-        SHARD=0 ;;
-      13b-Python)
-        SHARD=1 ;;
-      34b-Python)
-        SHARD=3 ;;
-      7b-Instruct)
-        SHARD=0 ;;
-      13b-Instruct)
-        SHARD=1 ;;
-      34b-Instruct)
-        SHARD=3 ;;
-      *)
-        echo "Unknown model: $m"
+# Function to check for prerequisites
+check_prerequisites() {
+    missing=false
+    command -v wget >/dev/null 2>&1 || { missing=true; echo "wget is missing."; }
+    command -v md5sum >/dev/null 2>&1 || { missing=true; echo "md5sum is missing."; }
+    
+    if [ "$missing" = true ]; then
+        echo "Would you like to install the missing prerequisites? (y/n)"
+        read install
+        if [ "$install" == "y" ]; then
+            sudo apt update
+            sudo apt install -y wget coreutils
+        else
+            echo "Exiting because prerequisites are not met."
+            exit 1
+        fi
+    fi
+}
+
+# Function to log messages with colors
+log() {
+    echo -e "$1"
+}
+
+# Function to download a file
+download_file() {
+    wget -c --quiet --show-progress $1 -O $2 &
```

Comment:
Should we do quiet? since the downloads could take hours at times, should we allow to show the progress?

Reply:
so now the download_file() will operate based on verbosity choice and download_model has been modified to capture this as well